src/java.base/share/native/libjli/java.c

```diff
@@ -72,6 +77,46 @@
 
 #define USE_STDERR JNI_TRUE     /* we usually print to stderr */
 #define USE_STDOUT JNI_FALSE
+#if defined(J9VM_OPT_CRAC_SUPPORT)
+#define EQUALS_SIGN '='
+#define NULL_TERMINATOR '\0'
+#define OPTION_VALUE_MEMORY_ALLOCATION_ERROR_MESSAGE "Failed to allocate memory for the value of the command line option %s."
+#define OPTION_VALUE_NOT_FOUND_ERROR_MESSAGE "The value of the command line option %s was not found."
+#define LOG_LEVEL_LENGTH_CALCULATATION_ERROR_MESSAGE "Failed to calculate the length of the CRIU log level string."
+#define LOG_LEVEL_MEMORY_ALLOCATION_ERROR_MESSAGE "Failed to allocate memory for the CRIU log level."
+#define OPTION_VALUE_NOT_VALID_ERROR_MESSAGE "The value of the command line option is not valid, option=%s, value=%s."
+#define OPTION_VALUE_NOT_EXPECTED_ERROR_MESSAGE "The value of the command line option is not expected, option=%s, value=%s."
+#define LOG_FILE_LENGTH_CALCULATION_ERROR_MESSAGE "Failed to calculate the length of the CRIU log file string."
+#define LOG_FILE_MEMORY_ALLOCATION_ERROR_MESSAGE "Failed to allocate memory for the CRIU log file option."
+#define CHECKPOINT_DIRECTORY_ERROR_MESSAGE "Failed to get the CRIU checkpoint directory, error=%d."
+#define LOG_LEVEL_ERROR_MESSAGE "Failed to get the CRIU log level, error=%d."
+#define UNPRIVILEGED_MODE_ERROR_MESSAGE "Failed to get the CRIU unprivileged mode, error=%d."
+#define LOG_FILE_ERROR_MESSAGE "Failed to get the CRIU log file, error=%d."
+#define RESTORE_FROM_CHECKPOINT_ERROR_MESSAGE "Failed to restore from checkpoint."
+#define RESTORE_CHILD_PROCESS_FAILED_ERROR_MESSAGE "The CRIU restore child process failed."
+#define MEMORY_ALLOCATION_ERROR 1
+#define OPTION_NAME_NOT_FOUND_ERROR 2
+#define OPTION_VALUE_NOT_FOUND_ERROR 3
+#define OPTION_VALUE_NOT_VALID_ERROR 4
+#define OPTION_VALUE_NOT_EXPECTED_ERROR 5
+#define CRAC_RESTORE_FROM_OPTION_NAME "-XX:CRaCRestoreFrom"
+#define LOG_LEVEL_OPTION_NAME "-Dopenj9.internal.criu.logLevel"
+#define LOG_LEVEL_OPTION_FORMAT "-v%d"
+#define UNPRIVILEGED_MODE_OPTION_NAME "-Dopenj9.internal.criu.unprivilegedMode"
+#define LOG_FILE_OPTION_NAME "-Dopenj9.internal.criu.logFile"
+#define LOG_FILE_OPTION_FORMAT "--log-file=%s"
+#define CRIU_RESTORE_MIN_ARGS 6
+#define CRIU_RESTORE_MAX_ARGS 9
+#define CRIU_COMMAND "criu"
+#define CRIU_RESTORE_OPTION "restore"
+#define CRIU_CHECKPOINT_DIRECTORY_OPTION "-D"
+#define CRIU_DEFAULT_LOG_LEVEL_OPTION "-v2"
+#define CRIU_UNPRIVILEGED_MODE_OPTION "--unprivileged"
+#define CRIU_SHELL_JOB_OPTION "--shell-job"
+#define EXECVP_ERROR -1
+#define EXECVP_SUCCESS 0
+#define WAIT_INDEFINITELY 0
```

Comment:
I'd like to see more order here: the names are generic and should be chosen so their intent, classification and usage should be clear.

Reply:
@singh264 please address this and the following comments.

Reply:
It seems like I will address this and the following comments next week.